File: frontend/lib/models/pedido.dart
class Pedido {
  String id;
  int nPedido;
  String detallesPedido;
  String estadoPedido;
  double precioTotal;
  String? usuarioId;
  String? nombreUsuario;
  String? nombreCompleto;
  String? direccion;
  String? ciudad;
  String? codigoPostal;
  String? telefono;
  String? email;
  String? comentarios;

  String get estado => estadoPedido;
  double get total => precioTotal;
  String? get usuario => nombreCompleto ?? nombreUsuario ?? usuarioId;
  List<dynamic> get productos => [];

  Pedido({
    required this.id,
    required this.nPedido,
    required this.detallesPedido,
    required this.estadoPedido,
    required this.precioTotal,
    this.usuarioId,
    this.nombreUsuario,
    this.nombreCompleto,
    this.direccion,
    this.ciudad,
    this.codigoPostal,
    this.telefono,
    this.email,
    this.comentarios,
  });

  String get getId => id;
  int get getNPedido => nPedido;
  String get getDetallesPedido => detallesPedido;
  String get getEstadoPedido => estadoPedido;
  double get getPrecioTotal => precioTotal;
  String? get getUsuarioId => usuarioId;
  String? get getNombreUsuario => nombreUsuario;
  String? get getNombreCompleto => nombreCompleto;
  String? get getDireccion => direccion;
  String? get getCiudad => ciudad;
  String? get getCodigoPostal => codigoPostal;
  String? get getTelefono => telefono;
  String? get getEmail => email;
  String? get getComentarios => comentarios;

  set setId(String id) => this.id = id;
  set setNPedido(int nPedido) => this.nPedido = nPedido;
  set setDetallesPedido(String detallesPedido) =>
      this.detallesPedido = detallesPedido;
  set setEstadoPedido(String estadoPedido) => this.estadoPedido = estadoPedido;
  set setPrecioTotal(double precioTotal) => this.precioTotal = precioTotal;
  set setUsuarioId(String? usuarioId) => this.usuarioId = usuarioId;
  set setNombreUsuario(String? nombreUsuario) =>
      this.nombreUsuario = nombreUsuario;
  set setNombreCompleto(String? nombreCompleto) =>
      this.nombreCompleto = nombreCompleto;
  set setDireccion(String? direccion) => this.direccion = direccion;
  set setCiudad(String? ciudad) => this.ciudad = ciudad;
  set setCodigoPostal(String? codigoPostal) => this.codigoPostal = codigoPostal;
  set setTelefono(String? telefono) => this.telefono = telefono;
  set setEmail(String? email) => this.email = email;
  set setComentarios(String? comentarios) => this.comentarios = comentarios;

  set estado(String value) => estadoPedido = value;

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nPedido': nPedido,
      'detallesPedido': detallesPedido,
      'estadoPedido': estadoPedido,
      'precioTotal': precioTotal,
      'usuarioId': usuarioId != null && usuarioId!.isNotEmpty
          ? int.tryParse(usuarioId!)
          : null,
      'nombreUsuario': nombreUsuario,
      'nombreCompleto': nombreCompleto,
      'direccion': direccion,
      'ciudad': ciudad,
      'codigoPostal': codigoPostal,
      'telefono': telefono,
      'email': email,
      'comentarios': comentarios,
    };
  }

  factory Pedido.fromJson(Map<String, dynamic> json) {
    return Pedido(
      id: json['id'].toString(),
      nPedido: json['nPedido'] ?? 0,
      detallesPedido: json['detallesPedido'] ?? '',
      estadoPedido: json['estadoPedido'] ?? '',
      precioTotal: json['precioTotal'] ?? 0.0,
      usuarioId: json['usuarioId']?.toString(),
      nombreUsuario:
          json['nombreUsuario']?.toString() ?? json['usuario']?.toString(),
      nombreCompleto: json['nombreCompleto']?.toString(),
      direccion: json['direccion']?.toString(),
      ciudad: json['ciudad']?.toString(),
      codigoPostal: json['codigoPostal']?.toString(),
      telefono: json['telefono']?.toString(),
      email: json['email']?.toString(),
      comentarios: json['comentarios']?.toString(),
    );
  }
}

File: backend/src/main/java/com/wabizone/ecommerce/models/Pedido.java
package com.wabizone.ecommerce.models;
import java.util.Objects;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;


//Modelo objeto relacionandolo con la tabla 
@Entity
@Table(name = "Pedido")
public class Pedido {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "n_pedido")
    private Long nPedido;

    @Column(name = "detalles_pedido", columnDefinition = "LONGTEXT")
    private String detallesPedido;

    @Column(name = "estado_pedido")
    private String estadoPedido;
    
    @Column(name = "precio_total")
    private double precioTotal;

    @Column(name = "usuario_id")
    private Long usuarioId;

    @Column(name = "nombre_usuario")
    private String nombreUsuario;

    @Column(name = "nombre_completo")
    private String nombreCompleto;

    @Column(name = "direccion")
    private String direccion;

    @Column(name = "ciudad")
    private String ciudad;

    @Column(name = "codigo_postal")
    private String codigoPostal;

    @Column(name = "telefono")
    private String telefono;

    @Column(name = "email")
    private String email;

    @Column(name = "comentarios")
    private String comentarios;

    public Pedido() {
    }

    //Generado en source actions solo el id de la tabla para comprobar
    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Pedido other = (Pedido) obj;
        return Objects.equals(this.id, other.id);
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getNPedido() {
        return nPedido;
    }

    public void setNPedido(Long nPedido) {
        this.nPedido = nPedido;
    }

    public String getDetallesPedido() {
        return detallesPedido;
    }

    public void setDetallesPedido(String detallesPedido) {
        this.detallesPedido = detallesPedido;
    }

    public String getEstadoPedido() {
        return estadoPedido;
    }

    public void setEstadoPedido(String estadoPedido) {
        this.estadoPedido = estadoPedido;
    }

    public double getPrecioTotal() {
        return precioTotal;
    }

    public void setPrecioTotal(double precioTotal) {
        this.precioTotal = precioTotal;
    }

    // Getters y setters para campos de usuario
    public Long getUsuarioId() {
        return usuarioId;
    }

    public void setUsuarioId(Long usuarioId) {
        this.usuarioId = usuarioId;
    }

    public String getNombreUsuario() {
        return nombreUsuario;
    }

    public void setNombreUsuario(String nombreUsuario) {
        this.nombreUsuario = nombreUsuario;
    }

    // Getters y setters para campos de envío
    public String getNombreCompleto() {
        return nombreCompleto;
    }

    public void setNombreCompleto(String nombreCompleto) {
        this.nombreCompleto = nombreCompleto;
    }

    public String getDireccion() {
        return direccion;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public String getCiudad() {
        return ciudad;
    }

    public void setCiudad(String ciudad) {
        this.ciudad = ciudad;
    }

    public String getCodigoPostal() {
        return codigoPostal;
    }

    public void setCodigoPostal(String codigoPostal) {
        this.codigoPostal = codigoPostal;
    }

    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getComentarios() {
        return comentarios;
    }

    public void setComentarios(String comentarios) {
        this.comentarios = comentarios;
    }
}
